### PR TITLE
Transaction queue tests update 

### PR DESCRIPTION
**Pull request by NikVolf**
This also adds some tests for maintaince

### CODE REVIEW

File: client/transaction-pool/src/tests/mod.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018-2020 Parity Technologies (UK) Ltd.
+// Copyright 2020 Parity Technologies (UK) Ltd.
```
Comment (tomusdrw) [inline client/transaction-pool/src/tests/mod.rs]: I think should stay 2018-2020, but I don't care enough :)

```suggestion
// Copyright 2018-2020 Parity Technologies (UK) Ltd.
```

File: client/transaction-pool/src/tests/api.rs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+pub struct ChainState {
+	pub block_by_number: HashMap<BlockNumber, Vec<Extrinsic>>,
+	pub block_by_hash: HashMap<Hash, Vec<Extrinsic>>,
+	pub header_by_number: HashMap<BlockNumber, Header>,
+	pub nonces: HashMap<AccountId, u64>,
+	pub invalid_hashes: HashSet<Hash>,
+}
+
+pub struct TestApi {
+	pub valid_modifier: RwLock<Box<dyn Fn(&mut ValidTransaction) + Send + Sync>>,
```
Comment (tomusdrw) [inline client/transaction-pool/src/tests/api.rs]: Maybe better add a method to alter that and keep all the fields private? Now it's a bit inconsistent.

File: client/transaction-pool/src/tests/api.rs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+	chain: RwLock<ChainState>,
+	validation_requests: RwLock<Vec<Extrinsic>>,
+}
+
+impl TestApi {
+	pub fn default() -> Self {
```
Comment (tomusdrw) [inline client/transaction-pool/src/tests/api.rs]: Should be `Default` impl, no?

File: client/transaction-pool/src/tests/api.rs [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+		self.chain.write().invalid_hashes.insert(
+			Self::hash_and_length_inner(xts).0
+		);
+	}
+
+	pub fn validation_requests(&self) -> Vec<Extrinsic> {
+		self.validation_requests.read().clone()
+	}
+
+	pub fn inc_nonce(&self, account: AccountId) {
```
Comment (tomusdrw) [inline client/transaction-pool/src/tests/api.rs]: I think I'd prefer `increment_nonce`, but not super strong on it.

File: client/transaction-pool/src/tests/api.rs (hunk 3 above)
Comment (NikVolf) [reply client/transaction-pool/src/tests/api.rs]: I renamed it to empty, there was a semantic change there also

File: client/transaction-pool/src/tests/mod.rs (hunk 1 above)
Comment (NikVolf) [reply client/transaction-pool/src/tests/mod.rs]: It's just GitHub thinks the file moved, while it is in fact a new file completely

**Pull request merged**